### DISCUSSION TITLE
How to specify a loader globally?

### DISCUSSION THREAD

**Discussion by ernestostifano** [remix-run/remix#5872] [Q&A]
Hello everyone!

I've been thinking to use Route's "loader" feature to check current user permissions against a map. If the user has permissions to access a specific route, the loader will return `null` gracefully and the corresponding `element` will be rendered normally. If permission check fails, the loader would throw an error, making the `errorElement` handle the situation.

At least theoretically, every should work, but this require me to specify that general loader for each one of the routes. Is there a way to pass it as a "global" loader? So it will be used for all routes?

If it is not possible, how about making it a feature? Maybe we could "chain" global and "local" loaders (and even accept an array of loaders at route level), so the result of the previous loader would be passed to the next until there are no mo loaders left. This could be implemented without making breaking changes.

Beside this, I haven't written the code yet, do you know if loaders can be synchronous?

Thanks a lot

**Comment by lili21** [comment]
You can do it with express middleware.  pick the express template.
